cannot have same jvm and webserver name validation

rahulsayini opened the pull request:
Before you contribute, please review these guidelines to help ensure a smooth process for everyone.

Thanks.

* Read [how to properly contribute to open source projects on Github][1].
* Fork the project.
* Use a feature branch.
* Write [good commit messages][2].
* Use the same coding conventions as the rest of the project.
* Commit locally and push to your fork until you are happy with your contribution.
* Make sure to add tests and verify all the tests are passing when merging upstream.
* Add an entry to the [Changelog][3] accordingly.
* Please add your name to the [CONTRIBUTORS.md][7] file. Adding your name to the [CONTRIBUTORS.md][7] file signifies agreement to all rights and reservations provided by the [License][4].
* [Squash related commits together][5].
* Open a [pull request][6].
* The pull request will be reviewed by the community and merged by the project committers.

[1]: http://gun.io/blog/how-to-github-fork-branch-and-pull-request
[2]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[3]: ./CHANGELOG.md
[4]: ./LICENSE
[5]: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
[6]: https://help.github.com/articles/using-pull-requests
[7]: ./CONTRIBUTORS.md

rahulsayini committed:
cannot have same jvm and webserver name validation

rahulsayini committed:
cannot have same jvm and webserver name validation

akinny commented on jwala-services/src/main/java/com/cerner/jwala/service/webserver/impl/WebServerServiceImpl.java:
```diff
@@ -84,6 +89,15 @@ private void initInMemoryStateService() {
                                      final User aCreatingUser) {
         createWebServerRequest.validate();
+        try {
+            if (null != jvmPersistenceService.findJvmByExactName(createWebServerRequest.getName())) {
```
I think we should do findByName for all

jmahmoodcc replied:
Remove the "if != null"

rahulsayini committed:
update postman resources collection

rahulsayini committed:
update postman resources collection

Sharvaribarve committed:
Added condition that throws exception for non-new State while renaming web servers

rahulsayini committed:
update postman resource files and collection

jmahmoodcc committed:
notify the UI that a web server was generated so the state changes from NEW to STOPPED

jmahmoodcc committed:
notify the UI that a web server was generated so the state changes from NEW to STOPPED

Sharvaribarve committed:
Changed position of enum, used message format and formatted code

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java:
```diff
@@ -120,9 +125,22 @@ public JvmServiceImpl(final JvmPersistenceService jvmPersistenceService,
 
+    private void validateWebserverNameConflict(CreateJvmRequest aCreateJvmRequest) {
+        try {
+            if (null != webServerPersistenceService.findWebServerByName(aCreateJvmRequest.getJvmName())) {
```
Remove the "if != null" around the call to findWebServerByName. Since the API is checking for an exact name change, if the NoResultException isn't thrown then there was a conflict.

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java:
```diff
@@ -251,6 +269,15 @@ public Jvm getJvm(final String jvmName) {
 
         updateJvmRequest.validate();
+        try {
+            if (null != webServerPersistenceService.findWebServerByName(updateJvmRequest.getNewJvmName())) {
```
Same thing: no need to use the "if != null", just call findWebServerByName

jmahmoodcc commented on jwala-services/src/main/java/com/cerner/jwala/service/webserver/impl/WebServerServiceImpl.java:
```diff
@@ -138,7 +152,15 @@ public WebServer getWebServer(final String aWebServerName) {
         anUpdateWebServerCommand.validate();
-
+        try {
+            if (null != jvmPersistenceService.findJvmByExactName(anUpdateWebServerCommand.getNewName())) {
```
Remove the "if != null"

jmahmoodcc committed:
Merge pull request #68 from cerner/renameWebServer

Throwing exception for non-new State upon renaming web-server

jmahmoodcc committed:
remove the messaging template fro the web server service and use the simple messaging service in the REST layer instead

jmahmoodcc committed:
remove the messaging template fro the web server service and use the simple messaging service in the REST layer instead

LinHungWu committed:
updating buildVersion for JWALA to 0.0.192 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.193-SNAPSHOT

jmahmoodcc committed:
make lock manager and message service private in WebServerServiceRestImpl

jmahmoodcc committed:
make lock manager and message service private in WebServerServiceRestImpl

jmahmoodcc committed:
Merge branch 'webserver-deploy' of github.com:cerner/jwala into webserver-deploy

jmahmoodcc committed:
Merge pull request #69 from cerner/webserver-deploy

notify the UI that a web server was generated

rahulsayini committed:
address code review comments

rahulsayini committed:
address code review comments

jeddcuison committed:
Check the jvm state first before checking if state is stale

This prevents unnecessary logging  that the JVM is stale (stopped JVMs are supposed to be inactive so there's no point in checking if their state is stale or not)

LinHungWu committed:
update jwala h2 with 1.3.176.jar

LinHungWu committed:
revert back to 1.3.175.jar

LinHungWu committed:
updating buildVersion for JWALA to 0.0.193 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.194-SNAPSHOT

rahulsayini committed:
remove unused code and add test cases to postman collection

LinHungWu committed:
updating buildVersion for JWALA to 0.0.194 for release

LinHungWu committed:
updating buildVersion to next snapshot version 0.0.195-SNAPSHOT

LinHungWu committed:
update h2 1.3.176.jar from .175

LinHungWu committed:
Merge branch 'master' of github.com:cerner/jwala

jeddcuison committed:
Put isStale at the end of the if statement

jmahmoodcc committed:
Merge pull request #70 from cerner/prevent-jvm-pinger-to-log-stopped-jvm-as-stale

Check the jvm state first before checking if state is stale

rahulsayini committed:
add service name validation for jvm and webserver create and update

rahulsayini committed:
add service name validation for jvm and webserver create and update

rahulsayini committed:
Merge branch 'service-name-conflict' of https://github.com/cerner/jwala into service-name-conflict

# Conflicts:
#	jwala-integration-test/src/test/postman/jwala-collection.postman_collection.json
#	jwala-persistence/src/main/java/com/cerner/jwala/persistence/service/impl/JpaJvmPersistenceServiceImpl.java

rahulsayini committed:
resolve master conflicts

rahulsayini committed:
resolve JpaJvmPersistenceServiceImpl.java conflicts

jeddcuison commented on jwala-persistence/src/main/java/com/cerner/jwala/persistence/service/impl/JpaJvmPersistenceServiceImpl.java:
```diff
@@ -257,16 +255,8 @@ public int removeTemplate(final String jvmName, final String templateName) {
     public boolean checkJvmResourceFileName(final String groupName, final String jvmName, final String fileName) {
         return jvmCrudService.checkJvmResourceFileName(groupName, jvmName, fileName);
     }
+<<<<<<< HEAD
```
Hi @rahulsayini ,

This will break the build. I think you were not able to resolve the conflict cleanly. Can we pair together and work on your branch to see what is happening ?

jeddcuison commented on jwala-common/src/test/java/com/cerner/jwala/commandprocessor/impl/jsch/JschCommandProcessorImplTest.java:
```diff
@@ -1,67 +0,0 @@
-package com.cerner.jwala.commandprocessor.impl.jsch;
```
This should be in another branch. The concept of branching is so that changes to a particular goal are grouped together and easier to digest.

jeddcuison commented on jwala-common/src/test/java/com/cerner/jwala/commandprocessor/impl/jsch/JschRequestProcessorImplTest.java:
```diff
@@ -1,52 +0,0 @@
-package com.cerner.jwala.commandprocessor.impl.jsch;
```
This should be in another branch

jeddcuison reviewed (changes requested):
We need to pair up and see what's happening with your branch. There are unresolved conflicts in the pull request that would break the build.
